File: src/test/resources/support-js/sub_handler.js
module.exports = {
    create: function() {
        var obj = {
            handle : function(event) {
                Packages.io.vertx.codetrans.MethodExpressionTest.event = event;
            },
            instanceHandler : function(handler) {
                handler("hello_instance");
            }
        };
        return obj;
  }, classHandler: function(handler) {
        handler("hello_class");
    }
};

File: src/test/resources/support/sub_handler.rb
module Support
  class SubHandler
    def initialize()
    end
    def handle(event)
      Java::IoVertxCodetrans::MethodExpressionTest.event = event
    end
    def instance_handler(handler)
      handler.handle 'hello_instance'
    end
    def self.create
      SubHandler.new
    end
    def self.class_handler(handler)
      handler.handle 'hello_class'
    end
  end
end
